tools/py/neb_interpolate.py

```diff
@@ -0,0 +1,422 @@
+    scipy = None
+    scipy_exception = e
+
+
+def spline_resample(q, nbeads_old, nbeads_new, k=3):
+    """Resamples the intermediate points along the spline so that
+    all points are equidistant by the spline arc length.
+
+    Arguments:
+        q           - beads.q[:]
+        k           - 1 or 3: order of the spline
+        nbeads_old  - number of beads in the input path
+        nbeads_new      - desired number of beads
+    Returns:
+        new_q - resampled coordinates
+    """
+    from numpy.linalg import norm as npnorm
+
+    if scipy is None:
+        print("spline interpolation requires scipy module.")
+        raise (scipy_exception)
+
+    if nbeads_new < 2:
+        raise RuntimeError("nbeads_new < 2.")
+
+    # First, we calculate the current parameterization of the path
+    # according to 3N-D Euclidean distances between adjacent beads.
+    t = [0.0]
+    current_t = 0.0
+    print("Cartesian 3N-D distances between old beads:")
+    for i in range(1, nbeads_old):
+        dist = npnorm(q[i] - q[i - 1])
+        print("\tfrom %3d to %3d : %.6f bohr" % (i - 1, i, dist))
+        if dist <= 1e-2:
+            print("Warning: two adjacent beads are very close.")
+        current_t += dist
+        t.append(current_t)
+
+    t = np.array(t)
+    t /= t[-1]
+    print("Spline parameter 't' before resampling:")
+    print(t)
+    print("t[i] - t[i-1]:")
+    print(t - np.roll(t, 1))
+
+    # New parameterization, equidistant by spline parameter
+    new_t = np.arange(nbeads_new) / (nbeads_new - 1)
+    print("New t:")
+    print(new_t)
+
+    # This reshapes q to a list of 3*Natoms sets consisting of nbeads_old numbers each,
+    # describing trajectory of one Cartesian component of an atom over the path.
+    components = q.T
+
+    new_components = []
+    for comp in components:
+        # Interpolate the trajectory of one cartesian component by a spline.
+        # bc_type='natural' imposes zero second derivative at the endpoints.
+        if k == 3:
+            spl = make_interp_spline(t, comp, k=k, bc_type="natural")
+        elif k == 1:
+            spl = make_interp_spline(t, comp, k=k)
+        # Sample new parameter values uniformly from the old range.
+        # Resample positions of this atom.
+        new_components.append(splev(new_t, spl))
+
+    # Reshape the path back to the beads shape
+    new_q = np.array(new_components).T
+
+    print("Cartesian 3N-D distances between NEW beads:")
+    for i in range(1, nbeads_new):
+        dist = npnorm(new_q[i] - new_q[i - 1])
+        print("\tfrom %3d to %3d : %.6f bohr" % (i - 1, i, dist))
+
+    return new_q
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""A script for interpolating a NEB path"""
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        type=str,
+        default="endpoints",
+        help="Mode: 'endpoints' to interpolate between 2 geometries provided separately,\n"
```

Comment:
can you add 
choices=["endpoints", "xyz","chk"]?
It is simpler than checking that the provided mode is valid down the road.